src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2330,12 +2351,15 @@ private bool ConsumeSingleLineCommentMultiSegment(ReadOnlySpan<byte> localBuffer
                     {
                         idx++;
                     }
-                    HasValueSequence = true;
                     break;
                 }
                 idx = localBuffer.IndexOfAny(JsonConstants.LineFeed, JsonConstants.CarriageReturn);
                 if (idx != -1)
                 {
+                    if (idx > 0)
+                    {
+                        end = new SequencePosition(_currentPosition.GetObject(), _currentPosition.GetInteger() + _consumed + leftOver + idx);
```

Comment:
This arises because we want to exclude the line separators `\r`, `\n` or `\r\n` in the comment. Two cases arise,

1. Comment text and the entire line separator are included in one span. This is the above case of `idx > 0` in which line separator will never be at 0th byte. Hence we have to adjust `end` _into_ to the current span.
2. Comment text ends in previous span and its line separator is in current span. In such case the parsing will enter this block but `idx` will be `0`. Given that we are only interested in comment text (and don't want line separator), the `end` has to be in previous span which is already set at line [2391](https://github.com/dotnet/corefx/pull/35934/files#diff-ed80dc6b290b6833c7ed8386409ca144R2391).

Reply:
I have reworked this entire thing from scratch, as we discussed. Please compare the code in latest PR directly with original unmodified code.